src/scripts/main.js

```diff
@@ -1,3 +1,37 @@
 'use strict';
 
 // write code here
+const table = document.querySelector('table');
+
+const theads = [...table.tHead.rows[0].cells];
+const tRows = [...table.tBodies[0].rows];
+
+for (const thead of theads) {
+  thead.addEventListener('click', sortByColumn);
+}
+
+function sortByColumn(e) {
+  const idx = theads.findIndex((elem) => elem === e.currentTarget);
+
+  tRows.sort((a, b) => {
+    let first = a.cells[idx].innerText;
+    let second = b.cells[idx].innerText;
+
+    if (first.startsWith('$')) {
+      first = formatSalary(first);
+      second = formatSalary(second);
+
+      return first - second;
+    }
+
+    return first.localeCompare(second);
```

Comment:
The sorting function currently handles currency values correctly by converting them to numbers. However, if there are other numerical values in the table that are not formatted as currency, they will be sorted as strings. Consider adding logic to detect and handle such numerical values appropriately.